### PR TITLE
Issue 3 union support

### PR DESCRIPTION
**Pull request by dswitzer**
Fixes issue #3 by adding support for UNION and UNION ALL. You can either feed in another QB to the union() and unionAll() methods or use a closure to define the query.

### CODE REVIEW

File: models/Grammars/BaseGrammar.cfc [hunk 1]
```diff
@@ -517,6 +517,37 @@ component displayname="Grammar" accessors="true" {
...
+    */
+    private string function compileUnions( required QueryBuilder query, required array unions ) {
+        if ( arguments.unions.isEmpty() ) {
+            return "";
+        }
+        
+        var sql = arguments.unions.map(function (union){
+           var sql = arguments.union.query.toSQL();
+
+            if( arguments.union.query.getOrders().len() ){
```
Comment (elpete) [inline models/Grammars/BaseGrammar.cfc]: Let's do this check first so we don't waste time converting the SQL if there are order bys.

File: models/Query/QueryBuilder.cfc [hunk 2]
```diff
@@ -1259,6 +1267,47 @@ component displayname="QueryBuilder" accessors="true" {
...
 
+    /*******************************************************************************\
+    |         UNION functions                                                       |
+    \*******************************************************************************/
+
+    /**
+    * Add a UNION statement to the SQL.
+    *
+    * @input    Either a QueryBuilder instance or a closure to define the derived query.
+    * @all Determines if UNION statement should be a "UNION ALL".  Passing this as an argument is discouraged.  Use the dedicated `unionAll` where possible.
```
Comment (elpete) [inline models/Query/QueryBuilder.cfc:1278]: I like the note about this argument being discouraged in favor of `unionAll` 👍

**Review by elpete — CHANGES_REQUESTED** [review]

**Comment by elpete** [comment]
Also, can you please add this to the docs: https://github.com/ortus-docs/qb-docs

File: models/Grammars/BaseGrammar.cfc (hunk 1 above)
Comment (dswitzer) [reply models/Grammars/BaseGrammar.cfc]: I've made the change.

**Comment by dswitzer** [comment]
Shouldn't we wait to document the changes once the code is merged?

**Comment by elpete** [comment]
I’ll merge both when they exist.

**Comment by dswitzer** [comment]
I've put up a pull request for the UNION docs:

https://github.com/ortus-docs/qb-docs/pull/13

**Review by elpete — APPROVED** [review]

**Pull request merged**